test/spec/Feature/Query/AggregateFunctionsSpec.hs

```diff
@@ -152,6 +152,131 @@ allowed =
+              {"id": 3, "count": 3},
+              {"id": 4, "count": 1}]|]
+            { matchHeaders = [matchContentTypeJson] }
+          get "/process_supervisor?select=...supervisors(supervisor:id,supervisor_count:count())&order=supervisors(supervisor)" `shouldRespondWith`
+            [json|[
+              {"supervisor": 1, "supervisor_count": 2},
+              {"supervisor": 2, "supervisor_count": 2},
+              {"supervisor": 3, "supervisor_count": 3},
+              {"supervisor": 4, "supervisor_count": 1}]|]
+            { matchHeaders = [matchContentTypeJson] }
+        it "works on nested resources" $ do
+          get "/process_supervisor?select=supervisor_id,...processes(...process_costs(count()))&order=supervisor_id" `shouldRespondWith`
+            [json|[
+              {"supervisor_id": 1, "count": 2},
+              {"supervisor_id": 2, "count": 2},
+              {"supervisor_id": 3, "count": 2},
+              {"supervisor_id": 4, "count": 1}]|]
+            { matchHeaders = [matchContentTypeJson] }
+          get "/process_supervisor?select=supervisor:supervisor_id,...processes(...process_costs(process_costs_count:count()))&order=supervisor_id" `shouldRespondWith`
+            [json|[
+              {"supervisor": 1, "process_costs_count": 2},
+              {"supervisor": 2, "process_costs_count": 2},
+              {"supervisor": 3, "process_costs_count": 2},
+              {"supervisor": 4, "process_costs_count": 1}]|]
+            { matchHeaders = [matchContentTypeJson] }
+        it "works on nested resources grouped by spreaded fields" $ do
+          get "/process_supervisor?select=...processes(factory_id,...process_costs(count()))&order=processes(factory_id)" `shouldRespondWith`
+            [json|[
+              {"factory_id": 1, "count": 2},
+              {"factory_id": 2, "count": 4},
+              {"factory_id": 3, "count": 1}]|]
+            { matchHeaders = [matchContentTypeJson] }
+          get "/process_supervisor?select=...processes(factory:factory_id,...process_costs(process_costs_count:count()))&order=processes(factory)" `shouldRespondWith`
+            [json|[
+              {"factory": 1, "process_costs_count": 2},
+              {"factory": 2, "process_costs_count": 4},
+              {"factory": 3, "process_costs_count": 1}]|]
+            { matchHeaders = [matchContentTypeJson] }
+        it "works on different levels of the nested resources at the same time" $
+          get "/process_supervisor?select=...processes(factory:factory_id,processes_count:count(),...process_costs(process_costs_count:count()))&order=processes(factory)" `shouldRespondWith`
+            [json|[
+              {"factory": 1, "processes_count": 2, "process_costs_count": 2},
+              {"factory": 2, "processes_count": 4, "process_costs_count": 4},
+              {"factory": 3, "processes_count": 2, "process_costs_count": 1}]|]
+            { matchHeaders = [matchContentTypeJson] }
```

Comment:
This test illustrates why to use `COUNT(<row>)` instead of `COUNT(*)`. Here, there's a `process` that does not have a `process_cost` assigned. If we used `COUNT(*)` for the `process_costs` table, then the top level resource would not have an idea on which are the relevant columns to count from the intermediate resource (`processes`). Passing the full row solves this issue.